packages/block-library/src/image/edit.js

```diff
@@ -273,21 +273,24 @@ export function ImageEdit( {
 			return;
 		}
 
-		const file = getBlobByURL( url );
-
-		if ( file ) {
-			mediaUpload( {
-				filesList: [ file ],
-				onFileChange: ( [ img ] ) => {
-					onSelectImage( img );
-				},
-				allowedTypes: ALLOWED_MEDIA_TYPES,
-				onError: ( message ) => {
-					isTemp = false;
-					onUploadError( message );
-				},
+		// Do not use getBlobByURL as it will only work for blobs created by us.
+		// Pasting might give us a blob URL that is not in the cache.
+		window
+			.fetch( url )
+			.then( ( r ) => r.blob() )
```

Comment:
Should we even still have our own blob cache? Can we replace all usages with `fetch( url ).then( r => r.blob() )`? I don't know how our cache came to be.

Reply:
I think it was just a dumb thing made by me that then got put into its own package 🤦‍♀️ 
I can either remove deprecate the package entirely, or just take out getblobURL and create a new version that is async?

Reply:
I think I prefer to just deprecate the package entirely, it's pretty useless.

Reply:
@ellatrix: agreed, we should deprecate it and switch everything to standard interfaces.